airflow/models/taskinstancehistory.py

```diff
@@ -45,8 +47,50 @@ class TaskInstanceHistory(Base):
     run_id = Column(StringID(), nullable=False)
     map_index = Column(Integer, nullable=False, server_default=text("-1"))
     try_number = Column(Integer, nullable=False)
+    # TODO: duplicate TI table in TIhistory and use pre-commit hook to keep both in sync.
+    trigger_id = Column(Integer)
     # The rest of the columns are kept in sync with TaskInstance, added at the bottom of this file
 
+    dag_run = relationship(
```

Comment:
Having to keep these relationships in sync feels a little messy. I wonder if instead we should have a different response schema that cuts out all this extra stuff, and just returns the TI specific stuff.

Reply:
I think it’s ok since the relationships are just view only. A bit messy but we might need more queries if we have a separate schema?

Reply:
RTIF is the deciding factor I think. I don't want to imply that it's from the specific try. I think having the history endpoint return a subset of the data back, namely just thing that can actually vary by try, is reasonable.

Reply:
If we return only a subset of the data, the endpoint will be different from other endpoints in terms of the returned data.

Reply:
I also noticed that if we want to keep the attrs in both TI and TIs consistent with pre-commit, then we would need these relationships.